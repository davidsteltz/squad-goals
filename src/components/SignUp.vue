<template>
  <div class="auth-container">
    <h5 class="title">
      Sign Up For Activise
    </h5>
    <div class="card-panel">
    <img class="logo-image" src="../assets/activyselogo.png" alt="A step ladder logo with a blue background">
      <form name="contact">
          <div class="input-field col s6">
            <i class="material-icons prefix">account_circle</i>
            <input class="form-field" name="name" id="name" type="text" v-model="name" required/>
            <label class="form-label" for="name">
            Name:
            </label>
        </div>
          <div class="input-field col s6">
            <i class="material-icons prefix">mail_outline</i>
            <input class="form-field" name="email" id="email" v-model="email" type="text" required/>
            <label class="form-label" for="email">
            Email:
            </label>
        </div>
        <div class="input-field col s6">
            <i class="material-icons prefix">lock</i>
            <input class="form-field" name="password" id="password" type="password" v-model="password" required/>
            <label class="form-label" for="password">
            Password:
            </label>
        </div>
        <div class="input-field col s6">
            <i class="material-icons prefix">lock</i>
            <input class="form-field" name="confirmPassword" id="confirmPassword" type="password" v-model="confirmPassword" password/>
            <label class="form-label" for="confirmPassword">
            Confirm Password:
            </label>
        </div>
        <div class="input-field col s6">
            <label>
                <input id="indeterminate-checkbox" type="checkbox" v-model="ageConsent"/>
                <span>Over 13 Years Of Age?</span>
            </label>
        </div>

        <div class="submit">
            <button class="btn waves-effect waves-light" type="submit" name="action">Submit
                <i class="material-icons right">send</i>
            </button>
        </div>

      </form>
      <div class="">
            Already have an account? <a @click="toggleSignIn()">Login</a>
        </div>
    </div>
  </div>
</template>

<script>
export default {
  data: () => ({
    valid: true,
    name: "",
    email: "",
    password: "",
    confirmPassword: "",
    ageConsent: false
  })
};
</script>

<style scoped lang="scss">
.auth-container {
  width: 80%;
  height: 100%;
}

.title {
  margin: 30px 0 30px 0;
}

.logo-image {
  width: 25%;
  margin-top: -40px;
  @media (max-width: 750px) {
    width: 150px;
  }
}

.submit {
  margin: 70px 0 30px 0;
}
</style>