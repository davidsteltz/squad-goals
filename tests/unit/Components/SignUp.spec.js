import { mount, shallowMount } from '@vue/test-utils';
import SignUp from '@/components/SignUp.vue';

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(SignUp, {
    propsData: {},
    mocks: {},
    stubs: {},
    methods: {}
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('SignUp', () => {
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  test('has 5 input-field classes', () => {
    expect(wrapper.findAll('.input-field').length).toBe(5);
  });

  test('clicking sign in triggers toggle function', () => {
    wrapper.vm.toggleSignUp = jest.fn();
    const signInLink = '#sign-in-link';
    wrapper.find(signInLink).trigger('click');
    expect(wrapper.vm.toggleSignUp).toBeCalled();
  });

  test('renders correctly', () => {
    const fullRender = mount(SignUp);
    expect(fullRender.element).toMatchSnapshot();
  });
});

describe('SignUp', () => {
  it('toggles sign in link', () => {
    const wrapper = shallowMount(SignUp);
    const signUpLink = wrapper.find('#sign-in-link');
    signUpLink.trigger('click');
    // const title = wrapper.find('title')
    // expect(wrapper.text()).toMatch('Sign Up For Activise')
  });
});
