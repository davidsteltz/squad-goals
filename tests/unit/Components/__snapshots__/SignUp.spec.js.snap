// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<div
  class="auth-container"
>
  <h5
    class="title"
  >
    
    Sign Up For Activise
  
  </h5>
   
  <div
    class="card-panel"
  >
    <img
      alt="A step ladder logo"
      class="logo-image"
      src="../assets/activyselogo.png"
    />
     
    <form
      name="sign-up"
    >
      <div
        class="input-field col s6"
      >
        <i
          class="material-icons prefix"
        >
          account_circle
        </i>
         
        <input
          class="form-field validate"
          id="name"
          name="name"
          required="required"
          type="text"
        />
         
        <label
          class="form-label"
          for="name"
        >
          
          Name:
          
        </label>
      </div>
       
      <div
        class="input-field col s6"
      >
        <i
          class="material-icons prefix"
        >
          mail_outline
        </i>
         
        <input
          class="form-field validate"
          id="email"
          name="email"
          required="required"
          type="text"
        />
         
        <label
          class="form-label"
          for="email"
        >
          
          Email:
          
        </label>
      </div>
       
      <div
        class="input-field col s6"
      >
        <i
          class="material-icons prefix"
        >
          lock
        </i>
         
        <input
          class="form-field validate"
          id="password"
          name="password"
          required="required"
          type="password"
        />
         
        <label
          class="form-label"
          for="password"
        >
          
          Password:
          
        </label>
      </div>
       
      <div
        class="input-field col s6"
      >
        <i
          class="material-icons prefix"
        >
          lock
        </i>
         
        <input
          class="form-field validate"
          id="confirmPassword"
          name="confirmPassword"
          password=""
          type="password"
        />
         
        <label
          class="form-label"
          for="confirmPassword"
        >
          
          Confirm Password:
          
        </label>
      </div>
       
      <div
        class="input-field col s6"
      >
        <label>
          <input
            id="indeterminate-checkbox"
            type="checkbox"
          />
           
          <span>
            Over 13 Years Of Age?
          </span>
        </label>
      </div>
       
      <div
        class="submit"
      >
        <button
          class="btn waves-effect waves-light"
          name="action"
          type="submit"
        >
          Submit
              
          <i
            class="material-icons right"
          >
            send
          </i>
        </button>
      </div>
    </form>
     
    <div>
      
          Already have an account? 
      <a
        id="sign-in-link"
      >
        Login
      </a>
    </div>
  </div>
</div>
`;
